From: Milan Crha <mcrha@redhat.com>
Date: Tue, 29 Apr 2025 07:52:57 +0200
Subject: gs-update-monitor: Do not show "Updates are Out of Date" when
 allow-updates is disabled

Notifying users about out of date updates when there's set to not allow
updates in the GNOME Software does not make sense, thus avoid the notification
when the updates are not allowed.

Origin: upstream, 48.4, commit:c3bdf11ea828cc7dd96ef2159fcbf8e45800e245
Bug: https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2815
---
 src/gs-update-monitor.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/gs-update-monitor.c b/src/gs-update-monitor.c
index df5d8dc..2ef1b89 100644
--- a/src/gs-update-monitor.c
+++ b/src/gs-update-monitor.c
@@ -213,7 +213,8 @@ should_notify_about_pending_updates (GsUpdateMonitor *monitor,
 
 	if (apps == NULL || !gs_app_list_length (apps)) {
 		/* Notify only when the download is disabled, or cannot download, and it's the 4th day or it's more than 7 days */
-		if ((!should_download || !can_download) && (timestamp_days >= 7 || timestamp_days == 4)) {
+		if ((!should_download || !can_download) && (timestamp_days >= 7 || timestamp_days == 4) &&
+		     gs_plugin_loader_get_allow_updates (monitor->plugin_loader)) {
 			*out_title = _("Updates Are Out of Date");
 			*out_body = _("Please check for available updates");
 			res = TRUE;
